src/widget.cpp

```diff
@@ -1636,30 +1636,30 @@ NWidgetCore *NWidgetMatrix::GetWidgetFromPos(int x, int y)
 	/* Get the appropriate offsets so we can draw the right widgets. */
 	NWidgetCore *child = dynamic_cast<NWidgetCore *>(this->head);
 	assert(child != nullptr);
-	int start_x, start_y, base_offs_x, base_offs_y;
+	uint start_x, start_y, base_offs_x, base_offs_y;
 	this->GetScrollOffsets(start_x, start_y, base_offs_x, base_offs_y);
 
-	int offs_y = base_offs_y;
-	for (int y = start_y; y < start_y + this->widgets_y + 1; y++, offs_y += this->widget_h) {
+	uint offs_y = base_offs_y;
+	for (uint y = start_y; y < start_y + this->widgets_y + 1; y++, offs_y += this->widget_h) {
 		/* Are we within bounds? */
 		if (offs_y + child->smallest_y <= 0) continue;
-		if (offs_y >= (int)this->current_y) break;
+		if (offs_y >= this->current_y) break;
 
 		/* We've passed our amount of widgets. */
 		if (y * this->widgets_x >= this->count) break;
 
-		int offs_x = base_offs_x;
-		for (int x = start_x; x < start_x + this->widgets_x + 1; x++, offs_x += rtl ? -this->widget_w : this->widget_w) {
+		uint offs_x = base_offs_x;
+		for (uint x = start_x; x < start_x + this->widgets_x + 1; x++, offs_x += rtl ? -this->widget_w : this->widget_w) {
 			/* Are we within bounds? */
 			if (offs_x + child->smallest_x <= 0) continue;
-			if (offs_x >= (int)this->current_x) continue;
+			if (offs_x >= this->current_x) continue;
 
 			/* Do we have this many widgets? */
-			int sub_wid = y * this->widgets_x + x;
+			uint sub_wid = y * this->widgets_x + x;
 			if (sub_wid >= this->count) break;
 
 			child->AssignSizePosition(ST_RESIZE, offs_x, offs_y, child->smallest_x, child->smallest_y, rtl);
-			child->SetLowered(this->clicked == sub_wid);
+			child->SetLowered(this->clicked == (int)sub_wid);
```

Comment:
You happen to know why `this->clicked` still needs to be signed? (did not check the code, it was easier to ask you :D)

Reply:
-1 is used as "nothing clicked". Technically speaking I suppose I could convert it to use UINT_MAX as none instead, but... eh